requests_hardened/client.py

```diff
@@ -49,4 +47,8 @@ def prepare_request(self, request: Request) -> PreparedRequest:
             )
             request.url = url
             request.headers = headers
-        return super().prepare_request(request)
+        prep = super().prepare_request(request)
+        if self._config.user_agent_override is not None:
+            prep.headers.update({"User-Agent": self._config.user_agent_override})
```

Comment:
Is it really needed to change headers after `super().prepare_request(request)`? If not, I suggest just changing `request.headers`

Reply:
Unfortunately, this will fail to set proper headers - that was a good point to add the test cases.

Reply:
Can you add a comment explaining why in the code then?

Reply:
Ok, found the issue - it was the matter of which headers needed to be updated